lib/Sema/NameBinding.cpp

```diff
@@ -36,60 +37,539 @@
 using namespace swift;
 
 //===----------------------------------------------------------------------===//
-// NameBinder
+// MARK: NameBinder and supporting types
 //===----------------------------------------------------------------------===//
 
 using ImportedModule = ModuleDecl::ImportedModule;
+using ImportedModuleDesc = SourceFile::ImportedModuleDesc;
 using ImportOptions = SourceFile::ImportOptions;
+using ImportFlags = SourceFile::ImportFlags;
+
+namespace {
+  /// Represents an import which the NameBinder knows exists, but which has not
+  /// yet had its options checked, module loaded, or cross-imports found.
+  ///
+  /// An UnboundImport may represent a physical ImportDecl written in the
+  /// source, or it may represent a cross-import overlay that has been found and
+  /// needs to be loaded.
+  struct UnboundImport {
+    /// If this UnboundImport directly represents an ImportDecl, the ImportDecl
+    /// it represents.
+    ///
+    /// This property should only be used to control where information is added
+    /// to the AST. Don't retrieve information directly from the ImportDecl; use
+    /// the other member variables.
+    NullablePtr<ImportDecl> ID;
+
+    SourceLoc importLoc;
+    ImportOptions options;
+    StringRef privateImportFileName;
+    Located<ImportKind> importKind;
+    ModuleDecl::AccessPathTy modulePath;
+    ModuleDecl::AccessPathTy declPath;
```

Comment:
It would be useful to document the difference between `modulePath` and `declPath`.

Reply:
I've documented all of these members.